packages/beacon-node/src/eth1/provider/eth1Provider.ts

```diff
@@ -62,6 +78,44 @@ export class Eth1Provider implements IEth1Provider {
       jwtSecret: opts.jwtSecretHex ? fromHex(opts.jwtSecretHex) : undefined,
       metrics: metrics,
     });
+
+    this.rpc.emitter.on(JsonRpcHttpClientEvent.RESPONSE, () => {
+      const oldState = this.state;
+      this.state = Eth1ProviderState.ONLINE;
+
+      if (oldState !== Eth1ProviderState.ONLINE) {
+        this.logger?.info("Eth1Provider is back online", {oldState, newState: this.state});
+      }
+    });
+
+    this.rpc.emitter.on(JsonRpcHttpClientEvent.ERROR, ({error}) => {
+      if (isErrorAborted(error)) {
+        this.state = Eth1ProviderState.ONLINE;
+      } else if ((error as unknown) instanceof HttpRpcError || (error as unknown) instanceof ErrorJsonRpcResponse) {
+        this.state = Eth1ProviderState.ERROR;
+      } else if (error && isFetchError(error) && HTTP_FATAL_ERROR_CODES.includes((error as FetchError).code)) {
+        this.state = Eth1ProviderState.OFFLINE;
+      } else if (error && isFetchError(error) && HTTP_CONNECTION_ERROR_CODES.includes((error as FetchError).code)) {
+        this.state = Eth1ProviderState.AUTH_FAILED;
+      }
+
+      if (this.state !== Eth1ProviderState.ONLINE) {
+        if (isOneMinutePassed()) {
```

Comment:
@nazarhussain can you please explain how this should behave

1. should it invoke the error log at most once a minute
2. or should it just invoke the error log once and never again, until there is a period of 1 minute where there are no errors?


See issue reported [on discord](https://discord.com/channels/593655374469660673/593655641445367808/1208096150557556779), we are not actually printing out any errors and just spam the logs with `info: Eth1Provider is back online oldState=ERROR, newState=ONLINE`.

I pushed a few changes that might partially solve this by fixing `createElapsedTimeTracker` function, see https://github.com/ChainSafe/lodestar/compare/unstable...nflaig/eth1-error-logging

Reply:
Another issue was reported to this, see [discord](https://discord.com/channels/593655374469660673/593655641445367808/1210095818925809724). Does not seem to happen if EL is still synincing but needs more investigation as they use [openexecution](https://github.com/tennisbowling/openexecution) and not directly connect to EL client.

Reply:
When ever `JsonRpcHttpClient` encounter an error, it bubbles up to this function. We log the the error once per minute because once the rpc encounter error it's usually because the RPC node is offline or some other issue then it's stay that way very long we end up tons of error logs with same error.

Reply:
> We log the the error once per minute

that's what it should do but the function has a bug, fixed in https://github.com/ChainSafe/lodestar/compare/unstable...nflaig/eth1-error-logging

The other issue I see is how we do the status logging, as single failed request will set the status to `ERROR` while a single success request sets it to `ONLINE`.

If some request succeed and some fail which is the cause if nethermind can't serve historical logs but it can serve other requests then what ends up happning is that the logs are spammed with

`info: Eth1Provider is back online oldState=ERROR, newState=ONLINE.`

It more of an issue of the intended behavior vs. the acutal behavior, if we fix the implementation it should be fine. I added more details in the issue https://github.com/ChainSafe/lodestar/issues/6469.

Reply:
I went over the discussions I had for it's implementation, that was the  inspiration or this change: 

> also maybe every minute log the error message as well if the eth1 is still down ... ( i.e. if we make the request and its still error and the last error display message > 60 seconds , log the message and update last error display time) 

So yes there is a bug in the implementation, but not in the `createElapsedTimeTracker`. That one seems to be good as what is intended for, it's just tracking time since first invocation. 

For `eth1` provider we should do via rate limiting helpers for rate limit error per minute.